### PR TITLE
fix: https with autocert

### LINKED ISSUE #8266: fix https with autocert

### PR DESCRIPTION
**Pull request by jeremyletang**
close #8266

### CODE REVIEW

File: datanode/gateway/server/server.go [hunk 1]
```diff
@@ -111,9 +111,21 @@ func (srv *Server) Start(ctx context.Context) error {
 		TLSConfig: tlsConfig,
 	}
 
+	var fallbacksrv *http.Server
 	if srv.cfg.REST.Enabled || srv.cfg.GraphQL.Enabled {
 		eg.Go(func() error {
 			if srv.srv.TLSConfig != nil {
+				if fallback != nil {
+					eg.Go(func() error {
+						fallbacksrv = &http.Server{Addr: ":http", Handler: fallback}
```
[the remark this answers is not in the document]
Comment (jeremyletang) [reply datanode/gateway/server/server.go:120]: yes. let's encrypt cannot work if the port 80 is not available to run the challenge. The autocert basically never worked as it's implemented now.

[the remark this answers is not in the document]
Comment (pscott31) [reply datanode/gateway/server/server.go:120]: it did! I watched it with my own eyes on my machine, though you had to run the gateway on port 443 or forward 443 to the gateway port. Happy to approve this as it seems harmless but I don't really understand what problem this fixes or how :)

**Review by pscott31 — APPROVED** [review]

**Comment by wwestgarth** [comment]
The second check box in the ticket says "test that it actually works", which I'm sure you did. So can you attach some of that evidence to the ticket or this PR?

**Comment by jeremyletang** [comment]
> The second check box in the ticket says "test that it actually works", which I'm sure you did. So can you attach some of that evidence to the ticket or this PR please?

@wwestgarth https://darling.network/graphql/playground

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**